### PR TITLE
Compat with OCaml 5.3.0~alpha1

### PR DESCRIPTION
**Pull request by Julow**

### CODE REVIEW

**Review by panglesd — APPROVED** [review]
Looks good to me!

File: test/xref2/module_type_of_extra.t/run.t [hunk 1]
```diff
@@ -37,28 +37,10 @@ More advanced uses of `module type of`, including using functors.
...
-  module type X4 = functor (A : sig type t end) -> sig type t end
-  module SubX : sig type t type u end
-  module type X5 =
-    sig
-      module Y : functor (A : sig type t end) -> sig module Z = SubX end
-      module type Z = functor (A : sig type t end) -> sig module Z = SubX end
-      module X' : sig type t = SubX.t end
-    end
-  module type X6 = sig type t end
+
```
Comment (jonludlam) [inline test/xref2/module_type_of_extra.t/run.t:40]: What went wrong here?

Comment (Julow) [reply test/xref2/module_type_of_extra.t/run.t:40]: Output is unstable due to the new functor syntax in module type exprs. I thought this output wasn't very interesting so I removed it.

Comment (jonludlam) [reply test/xref2/module_type_of_extra.t/run.t:40]: That's a shame - this was intended to be directly compared with the output from odoc (see https://github.com/ocaml/odoc/pull/1202/files/6119c82332b0dc5756a58fa208fd83b9b98e1108#diff-b3b2e29d50da342308f1468a32da8a0a9a40ca40c5b3c654521d8b891b38fab8R49)

**Comment by jonludlam** [comment]
Thanks!

**Pull request merged**